pkg/scanner/utils/utils.go

```diff
@@ -2,27 +2,49 @@ package utils
 
 import (
 	"fmt"
+	"strconv"
 	"strings"
 
+	"github.com/Masterminds/semver/v3"
+
 	"github.com/aquasecurity/fanal/types"
+
 	"github.com/aquasecurity/trivy/pkg/log"
-	"github.com/knqyf263/go-version"
 )
 
 var (
 	replacer = strings.NewReplacer(".alpha", "-alpha", ".beta", "-beta", ".rc", "-rc")
 )
 
-func MatchVersions(currentVersion *version.Version, rangeVersions []string) bool {
-	for _, p := range rangeVersions {
-		c, err := version.NewConstraint(replacer.Replace(p))
+func MatchVersions(currentVersion *semver.Version, rangeVersions []string) bool {
+	for _, v := range rangeVersions {
+		v = replacer.Replace(v)
+		constraintParts := strings.Split(v, ",")
+		for j := range constraintParts {
+			constraintParts[j] = FormatPatchVersion(constraintParts[j])
+		}
+		v = strings.Join(constraintParts, ",")
+		c, err := semver.NewConstraint(v)
 		if err != nil {
-			log.Logger.Debug("NewConstraint", "error", err)
-			return false
+			log.Logger.Error("NewConstraint", "error", err)
+			continue
 		}
-		if c.Check(currentVersion) {
+		// Validate a version against a constraint.
+		valid, msgs := c.Validate(currentVersion)
+		if valid {
 			return true
 		}
+		for _, m := range msgs {
+			// re-validate after removing the patch version
+			if strings.HasSuffix(m.Error(), "is a prerelease version and the constraint is only looking for release versions") {
+				if v2, err := semver.NewVersion(fmt.Sprintf("%v.%v.%v", currentVersion.Major(), currentVersion.Minor(), currentVersion.Patch())); err == nil {
```

Comment:
nit: it is wide, so what if we split the line?

```suggestion
				v2, err := semver.NewVersion(fmt.Sprintf("%v.%v.%v", 
					currentVersion.Major(), currentVersion.Minor(), currentVersion.Patch()))
				if err == nil {
```